libats-messaging-datatype/src/main/scala/com/advancedtelematic/libats/messaging_datatype/Messages.scala

```diff
@@ -146,16 +167,42 @@ object Messages {
 
   case class ImageStorageUsage(namespace: Namespace, timestamp: Instant, byteCount: Long)
 
-  final case class DeviceUpdateEvent(namespace: Namespace,
-                                     eventTime: Instant,
-                                     correlationId: CorrelationId,
-                                     deviceUuid: DeviceId,
-                                     updateStatus: DeviceUpdateStatus)
-
   @deprecated("use data type from libtuf_server", "v0.1.1-21")
   case class DeviceUpdateReport(namespace: Namespace, device: DeviceId, updateId: UpdateId, timestampVersion: Int,
                                 operationResult: Map[EcuIdentifier, OperationResult], resultCode: Int)
 
+  sealed trait DeviceUpdateEvent {
+    def namespace: Namespace
+    def eventTime: Instant
+    def correlationId: CorrelationId
+    def deviceUuid: DeviceId
+  }
+
+  final case class DeviceUpdateAvailable(
+      namespace: Namespace,
+      eventTime: Instant,
+      correlationId: CorrelationId,
+      deviceUuid: DeviceId
+  ) extends DeviceUpdateEvent
+
+  final case class DeviceUpdateCanceled(
+      namespace: Namespace,
+      eventTime: Instant,
+      correlationId: CorrelationId,
+      deviceUuid: DeviceId
+  ) extends DeviceUpdateEvent
+
+  final case class DeviceUpdateCompleted(
+      namespace: Namespace,
+      eventTime: Instant,
+      correlationId: CorrelationId,
+      deviceUuid: DeviceId,
+      result: InstallationResult,
+      ecuReports: Map[EcuIdentifier, EcuInstallationReport],
```

Comment:
We should switch this ECU reports `Map[EcuId]` to `Seq[Item]` as aktualizr does it now (to support installation items that don't involve ECUs). But this requires another migration in device-registry that currently uses `DeviceInstallationReport`.

Reply:
Since we have two pending migrations in dev-reg, I would feel safer if we open a ticket for this migration to do after the pending ones have run and are stable.
Also, what would be the format of `Item`?